#1018 Remove "Enroll Now" button fixes mitodl/mitxpro#1018

ahmed-arbisoft opened the pull request:
~~This PR removes the "Enroll Now" link/button from the LMS UI.~~

~~This PR adds a feature setting `COURSE_DEFAULT_INVITE_ONLY` that disables/removes "Enroll Now" buttons on the UI. The default value is `False` and can be changed from the environment.~~

Updates the templates to respect `course.invitation_only` and show the correct message where applicable (and remove the "Enroll Now" button as well).

### How to test this
Create a course and set `course.invitation_only` to `True`. Visit the course page directly, it should not show any button to enroll yourself (typically labelled "Enroll Now").

~~Screenshot is highlighted where the button/link was removed.~~
### Screenshot
Course About
![Screenshot from 2019-09-18 18-02-03](https://user-images.githubusercontent.com/45350418/65153040-85437c80-da42-11e9-9350-76c6087bace9.png)

Course Home
![Screenshot from 2019-09-18 18-27-11](https://user-images.githubusercontent.com/45350418/65153050-8aa0c700-da42-11e9-9779-bf39ad613c8f.png)

asadiqbal08 reviewed (approved):
overall LGTM 👍

ahmed-arbisoft commented:
@pdpinch if you could look this over that'd be great.

pdpinch commented:
Sorry we don't have a PR template for this repo, but I don't think there's a way for us to do that. 

Can you add a "fixes https://github.com/mitodl/mitxpro/issues/1018" to the description? I think that will signal to github to put in the cross links.

pdpinch commented:
did you see this https://github.com/mitodl/mitxpro/issues/1018#issuecomment-530858548 ?

If we have any hope of getting edX to accept this upstream, it will have to be behind a feature flag or a course setting. Can you take a look at this `invitation_only` course setting: https://github.com/edx/edx-platform/blob/48c32a4367aa2e797b848c85950dcbeea265a101/openedx/features/course_experience/views/course_home_messages.py#L150 and see if we can use that?

ahmed-arbisoft commented:
Ah, I made a mistake there. I'll update the implementation to go with a setting.

ahmed-arbisoft commented:
The redo includes changes from @tasawernawaz PR against `hawthorn.master` for implementing a default feature flag for course invite only value: https://github.com/mitodl/edx-platform/pull/126

The rest is the same implementation but now utilizing the value from the setting.

tasawernawaz reviewed (approved):
👍

pdpinch commented:
The site-wide feature setting is for the default value of the course setting. I think it should be enough to check the value of `courselike.invitation_only`

pdpinch commented:
Can you check to see where this setting interferes with our code for enrolling users from xPRO? If it's very difficult, we could wait and test on CI and RC, but it would be might be best to know earlier.

pdpinch commented on openedx/features/course_experience/views/course_home_messages.py:
```diff
@@ -124,7 +125,7 @@ def _register_course_home_messages(request, course, user_access, course_start_da
             title=Text(_('You must be enrolled in the course to see course content.'))
         )
-    if not user_access['is_anonymous'] and not user_access['is_staff'] and not user_access['is_enrolled']:
+    if not user_access['is_anonymous'] and not user_access['is_staff'] and not user_access['is_enrolled'] and not (settings.FEATURES.get('COURSE_DEFAULT_INVITE_ONLY') or course.invitation_only):
```
If `course.invitation_only` can we show the same message used on the course_about.html page? 

`<span class="register disabled">${_("Enrollment in this course is by invitation only")}</span>`

ahmed-arbisoft replied:
Updated

Commit:
#1018 Remove "Enroll Now" button

Revert "#1018 Remove "Enroll Now" button"

This reverts commit 6c17b61d4f923af2ea59c4b2a42f868f5c01d848.

#1018 Remove enroll now button

Site-wide feature removal and invitation_only message

ahmed-arbisoft commented:
@pdpinch I'll have to dig in a little bit more regarding if the `invitation_only` setting might interfere or not, however I have a hunch that the service user that we use for `edx_api_client` is supposed to be a staff member on the Open edX side so this shouldn't pose a problem (the workflow becomes: a staff member is using edX API to enroll a registered user). I'll give it a test on my end and let you know.
